### PR TITLE
Storage function to fetch storage type

### PR DESCRIPTION
**Pull request by gohil-jay**
Addresses issue faced while solving https://github.com/scikit-hep/boost-histogram/issues/157 issue.

Simply added a call function that returns storage type.

This is also needed to correctly create an empty copy an arbitrary histogram:

```python
new_hist = bh.Histogram(*old_hist.axes, storage=old_hist.storage_type())
```

Backward compatibility shim needed for a few releases since things like Hist already are having to use this; it needs to be officially public API.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,6 +125,7 @@ histograms can be plotted via any compatible library, such as [mplhep][].
   * `*=`: Multiply by a scaler (not all storages) (`hist * scalar` and `scalar * hist` supported too)
   * `/=`: Divide by a scaler (not all storages) (`hist / scalar` supported too)
   * `.kind`: Either `bh.Kind.COUNT` or `bh.Kind.MEAN`, depending on storage
+  * `.storageType`: Fetch the histogram storage type in string
```
Comment (henryiii) [inline README.md:128]:
```suggestion
  * `.storage_type`: Fetch the histogram storage type
```

File: src/boost_histogram/_internal/hist.py [hunk 2]
```diff
@@ -345,6 +345,9 @@ def view(
         """
         return _to_view(self._hist.view(flow))
 
+    def storageType(self) -> str:
+        return str(self._storage_type).rsplit(".", maxsplit=1)[-1][:-2]
```
Comment (henryiii) [inline src/boost_histogram/_internal/hist.py:349]:
```suggestion
        return cast(self, self._storage_type, Storage)
```

File: src/boost_histogram/_internal/hist.py [hunk 3]
```diff
@@ -345,6 +345,9 @@ def view(
         """
         return _to_view(self._hist.view(flow))
 
+    def storageType(self) -> str:
```
Comment (henryiii) [inline src/boost_histogram/_internal/hist.py:348]:
```suggestion
    def storageType(self) -> Storage:
```

File: src/boost_histogram/_internal/hist.py (hunk 2 above)
Comment (henryiii) [inline src/boost_histogram/_internal/hist.py:349]:
```suggestion
    _storage_type = storage_type
```

And rename `_storage_type` to `storage_type` (change elsewhere too).

**Comment by gohil-jay** [comment]
Now all left to do is to run this by @HDembinski !

**Comment by HDembinski** [comment]
Please use the review feature, to make sure I am getting the notification. I have no objections.

File: src/boost_histogram/_internal/hist.py [hunk 4]
```diff
@@ -594,7 +594,7 @@ def _axis(self, i: int = 0) -> Axis:
         return cast(self, self._hist.axis(i), Axis)
 
     @property
-    def _storage_type(self) -> Type[Storage]:
+    def storage_type(self) -> Type[Storage]:
         return cast(self, self._hist._storage_type, Storage)  # type: ignore[return-value]
```
Comment (henryiii) [inline src/boost_histogram/_internal/hist.py:598]:
```suggestion
        return cast(self, self._hist._storage_type, Storage)  # type: ignore[return-value]
        
    # Backward compat
    _storage_type = storage_type
```

**Comment by henryiii** [comment]
Looks like moving to windows-2022 hits a ICE. 🤦

File: src/boost_histogram/_internal/hist.py [hunk 5]
```diff
@@ -594,9 +594,12 @@ def _axis(self, i: int = 0) -> Axis:
...
 
+    # Backward compat
+    _storage_type = storage_type
+
```
Comment (HDembinski) [inline src/boost_histogram/_internal/hist.py:602]: This should raise a deprecationwarning.

**Pull request merged**